firmware/arduino/chessboard/uart.ino

```diff
@@ -0,0 +1,62 @@
+char incomingByte;
+char  buffer[5];
+int byteNum = -1; //-1 indicates that the start code hasn't been received
+
+//Wait for input
+void serialEvent2()
+{
+    //Loop through all available bytes
+    while (Serial2.available())
+    {
+        //Get byte
+        incomingByte = (char) Serial2.read();
+
+        //Reset buffer position
+        if (incomingByte == '~')
+        {
+            byteNum = 0;
+        }
+        //Add byte to buffer
+        else if (byteNum != -1)
+        {
+            buffer[byteNum] = incomingByte;
+        }
+
+        //Check if the buffer is full, process input
+        if (byteNum == 4)
+        {
+            //Reset buffer position
+            byteNum = -1;
+
+            //Process input
+            switch(buffer[0])
+            {
+                case 0:
+                    //move_straight(int startRow, int startCol, int endRow, int endCol)
+                    move_straight(buffer[1] - 'a', buffer[2] - '0', buffer[3] - 'a', buffer[4] - '0');
+                    break;
+                case 1:
+                    //move_to_graveyard(int startRow, int startCol, char color, char type)
+                    move_to_graveyard(buffer[1] - 'a', buffer[2] - '0', buffer[3], buffer[4]);
+                    break;
+                case 2:
+                    //Args: piece color (b/w), kingside or queenside (-/0) 
+                    //castle(char color, char side)
+                    castle(buffer[1], buffer[3]);
```

Comment:
You should also have something here about queenside versus kingside castling

Reply:
Right now I'm just passing the - or 0 to the castle function that we're going to create in the future. 

I was leaning towards keeping the communications part of the code as simple as possible and handling queenside/kingside and white/black in the movement function.

Do you or @LSaber36 have a preference as to where it's handled?

Reply:
I agree, we want to separate the logic from the moves if possible. That's my opinion at least.

Reply:
Also, all of the ESP32 code to be on one file. It's okay to make changes to the main code as long as it's in your branch, where it's safe.

Reply:
This code should focus on receiving the UART bytes first and foremost, with interpreting them after. The first step is storing the bytes in a byte array of a set size, which I believe we said was 6.

Reply:
I think putting the message parsing code in a separate place is a good idea. Also, I understand what you are doing now, thanks for clarifying.

Reply:
When you make the `castle` function, can you make it clear what you're doing? A simple comment saying exactly what you explained here would be perfect

Reply:
> Also, all of the ESP32 code to be on one file. It's okay to make changes to the main code as long as it's in your branch, where it's safe.

I think we're better off using multiple files to avoid merge conflicts. The Arduino IDE automatically combines them, and it won't cause any issues as long as we declare functions in the primary chessboard.ino file.

Reply:
I didn't know you can do that! That's pretty cool. I think it will make things cleaner if we can have things in separate files; will lead to 1) fewer merge conflicts and 2) easier to understand code IMO